### ISSUE TITLE
Example crashes in Unity

### ISSUE THREAD

**Issue by dhowe** [exodrifter/unity-rumor#46]
Trying to get your original example fully working in Unity, but get the error below when I click following 'Great!', before the apples and pears part (I'm likely doing something wrong):

````
ArgumentException: Cannot assign values to type Exodrifter.Rumor.Expressions.AddExpression!
Exodrifter.Rumor.Expressions.SetExpression.Evaluate (Exodrifter.Rumor.Engine.Rumor rumor) (at Assets/Libraries/Rumor/Expressions/SetExpression.cs:20)
Exodrifter.Rumor.Nodes.Statement+<Run>c__Iterator0.MoveNext () (at Assets/Libraries/Rumor/Nodes/Statement.cs:34)
Exodrifter.Rumor.Engine.StackFrame+<Run>c__Iterator0.MoveNext () (at Assets/Libraries/Rumor/Engine/StackFrame.cs:124)
Exodrifter.Rumor.Engine.Rumor+<ExecuteStack>c__Iterator1.MoveNext () (at Assets/Libraries/Rumor/Engine/Rumor.cs:307)
Exodrifter.Rumor.Engine.Rumor+<Run>c__Iterator0.MoveNext () (at Assets/Libraries/Rumor/Engine/Rumor.cs:285)
UnityEngine.SetupCoroutine.InvokeMoveNext (IEnumerator enumerator, IntPtr returnValueAddress) (at /Users/builduser/buildslave/unity/build/Runtime/Export/Coroutines.cs:17)

**Comment by dhowe** [comment]
(can share my unity project if useful)

**Comment by exodrifter** [comment]
It's alright, I was able to reproduce your error. Thanks for your bug report. I know it can be really troublesome to file these, so I really appreciate it.

It looks like I accidentally broke the `+=`, `-=`, `/=`, `*=` operators when I optimized the tokenizer in #40.

What happens is the compiler accidentally considers the following line:
```
$ pears += 1
```
as (psuedocode):
```
(pears + no-op) = 1
```
And of course, you can't assign the value `1` to an addition expression. I'm actually rehauling the compiler right now, but since that won't be done until early January I'll see if I can make a bug fix. For now, you can work around the issue by using `$ pears = pears + 1` instead of the other operators.

Alternatively, you can drop back to [version 1.0.0](https://github.com/exodrifter/unity-rumor/releases/tag/1.0.0), which is right before the optimization update. However, there have been a lot of bugfixes since then, so I wouldn't recommend it.

**Comment by dhowe** [comment]
Workaround works for me until next release, thanks!

**Comment by exodrifter** [comment]
This bug is now fixed in [1.1.1](https://github.com/exodrifter/unity-rumor/releases/tag/1.1.1).
